src/main/java/org/threadly/concurrent/wrapper/intercepter/ExecutorTaskIntercepter.java

```diff
@@ -10,31 +11,44 @@
 
 /**
  * <p>Class to wrap {@link Executor} pool so that tasks can be intercepted and either wrapped, or 
- * modified, before being submitted to the pool.  This abstract class needs to have 
- * {@link #wrapTask(Runnable)} overridden to provide the task which should be submitted to the 
- * {@link Executor}.  Please see the javadocs of {@link #wrapTask(Runnable)} for more details 
- * about ways a task can be modified or wrapped.</p>
+ * modified, before being submitted to the pool. This class can be passed a lamba in the 
+ * constructor, or {@link #wrapTask(Runnable)} can be overridden to provide the task which should 
```

Comment:
> or {@link #wrapTask(Runnable)} can be overridden

I would change this to:

> or you can extend this class and override the function {@link #wrapTask(Runnable)}

People should know that you must extend the class to override wrapTask, but I think it's worth having that key word in the docs.